app/javascript/tests/public_health/Export.test.js

```diff
@@ -0,0 +1,124 @@
+const dropdownOptions = [ 'Line list CSV', 'Sara Alert Format', 'Excel Export For Purge-Eligible Monitorees', 'Excel Export For All Monitorees', 'Custom Format...' ];
+
+function getExposureWrapper() {
+  return shallow(<Export all_monitorees_count={200} authenticity_token={authyToken} current_monitorees_count={32} custom_export_options={{}}
+    jurisdiction_paths={mockJurisdictionPaths} jurisdiction={mockJurisdiction1} query={mockQuery1} tabs={mockExposureTabs} />);
+}
+
+function getIsolationWrapper() {
+  return shallow(<Export all_monitorees_count={200} authenticity_token={authyToken} current_monitorees_count={32} custom_export_options={{}}
+    jurisdiction_paths={mockJurisdictionPaths} jurisdiction={mockJurisdiction1} query={mockQuery2} tabs={mockIsolationTabs} />);
+}
+
+function getInstance() {
+  return shallow(<Export all_monitorees_count={200} authenticity_token={authyToken} current_monitorees_count={32} custom_export_options={{}}
+    jurisdiction_paths={mockJurisdictionPaths} jurisdiction={mockJurisdiction1} query={mockQuery2} tabs={mockIsolationTabs} />).instance();
+}
+
+describe('Export', () => {
+  it('Properly renders all main components', () => {
+    const wrapper = getExposureWrapper();
+    expect(wrapper.find(DropdownButton).exists()).toBeTruthy();
+    expect(wrapper.find(Dropdown.Item).length).toEqual(5);
+    expect(wrapper.find(Dropdown.Divider).length).toEqual(1);
+    expect(wrapper.find(ConfirmExport).length).toEqual(4);
+    wrapper.find(ConfirmExport).forEach(function(component) {
+      expect(component.prop('show')).toBeFalsy();
+    });
+    expect(wrapper.find(ToastContainer).exists()).toBeTruthy();
+    expect(wrapper.find(CustomExport).exists()).toBeFalsy();
+  });
+
+  it('Properly renders dropdown in exposure workflow', () => {
+    const wrapper = getExposureWrapper();
+    dropdownOptions.forEach(function(option, index) {
+      if (index < 2) {
+        option += ' (exposure)';
+      }
+      expect(wrapper.find(Dropdown.Item).at(index).text()).toEqual(option);
+    });
+  });
+
+  it('Properly renders dropdown in isolation workflow', () => {
+    const wrapper = getIsolationWrapper();
+    dropdownOptions.forEach(function(option, index) {
+      if (index < 2) {
+        option += ' (isolation)';
+      }
+      expect(wrapper.find(Dropdown.Item).at(index).text()).toEqual(option);
+    });
+  });
+
+  it('Clicking "Line list CSV" option displays Confirm Export modal', () => {
+    const wrapper = getExposureWrapper();
+    expect(wrapper.find(Dropdown.Item).at(0).text().includes(dropdownOptions[0])).toBeTruthy();
+    expect(wrapper.find(ConfirmExport).at(0).prop('show')).toBeFalsy();
+    wrapper.find(Dropdown.Item).at(0).simulate('click');
+    expect(wrapper.find(ConfirmExport).at(0).prop('show')).toBeTruthy();
+  });
+
+  it('Clicking "Sara Alert Format" option displays Confirm Export modal', () => {
```

Comment:
I think maybe for each of these where you are checking the Confirm Export modal, you should verify the `title` and `onStartExport` props are correct (since that would be a big bug if there was a copy/paste or typo that caused those to be incorrect).

Reply:
there isn't really a way to test the `onStartExport` because it's a function.  typically you don't test that props are correct so I'm not sure we should add a title check but up to you

Reply:
We shouldn't be testing the full functionality of the `submit` function in this specific unit test here (though we should elsewhere in these tests), but I think right now this tests that it renders a confirm export modal, but it doesn't test that it's actually the right one and will trigger the right export, which is one of the most important parts. It's the job of the `Export` component to pass the correct props to the the `ConfirmExport` component, so that's why I'd like them to be tested. So yes definitely think we should at least test that we are passing the correct title, and I do still think we can and should test we're passing the correct function function call as well. Believe we can do that by calling the child prop and verifying it calls submit with the correct string. Should do the same for `show` and `onCancel` props as well. Thoughts?

Reply:
the show prop is already being tested.  I added a test for the title top.  there's no way to test what is being passes into the submit function unfortunately.  Same with on cancel.   I could break that URL into a separate prop and test it that way but I don't love that

Reply:
See here:
![Screen Shot 2020-12-31 at 11 56 03 AM](https://user-images.githubusercontent.com/35042815/103419184-ecb20c80-4b5f-11eb-81dc-ceb166c44f16.png)

Reply:
It's also not possible to test the submit functionality because this is strictly a component test that doesn't have access to the server.  That's something that would need to be added as a part of the system tests

Reply:
update, I did a slight refactor that allows for testing the title and onStartExport params. lmk what you think

Reply:
This is much better, and a good example of a time when something being hard to test indicated a implementation design change was probably needed. Thank you!